src/ambianic/pipeline/ai/fall_detect.py

```diff
@@ -76,22 +79,42 @@ def process_sample(self, **sample):
 
 
     def calculate_angle(self, p):
-        x1, y1 = p[0][0][0], p[0][0][1]
-        x2, y2 = p[0][1][0], p[0][1][1]
-        x3, y3 = p[1][0][0], p[1][0][1]
-        x4, y4 = p[1][1][0], p[1][1][1]
+
+        angle = 0
+
+        body_line_detected = len(p[0]) == len(p[1]) == 2
+        if not body_line_detected:
+            return angle
+        
+        x1, y1 = p[0][0]
+        x2, y2 = p[0][1]
+    
+        x3, y3 = p[1][0]
+        x4, y4 = p[1][1]
 
         theta1 = math.degrees(math.atan2(y2-y1, x1-x2))
         theta2 = math.degrees(math.atan2(y4-y3, x3-x4))
             
-        theta = abs(theta1-theta2)
-        return theta
+        angle = abs(theta1-theta2)
+        return angle
+
+
+    def is_body_line_motion_downward(self, left_angle_with_yaxis, rigth_angle_with_yaxis):
```

Comment:
@bhavikapanara this is a nice , short, clean and descriptive method. Does not need extra comments. Nice reference point for other methods.

Reply:
thanks @ivelin